### PR TITLE
Cosmetic changes to directtools plot on Workbench

### LINKED ISSUE #27650: Cosmetic changes to directtools plot on Workbench
As Workbench uses matplotlib, the behaviour of some plot functions within the directtools module have changed with regard to Mantidplot.

### Requested changes

1. Thinner lines for the plots. 

2. Add end caps to the error bars (horizontal bars at the end of the vertical bars). 

3. Add tick marks on the top scale and on the right-hand side. 
   This also applies to the maps.

4. Put the tick marks on the inside rather than on the outside.

### PR DESCRIPTION
**Pull request by MathieuTillet**
**Description of work.**

This adds flags to some plots in the directtools module, so that the figures plotted using Workbench are as similar as possible to what they looked like in Mantidplot.

More precisely, it : 
- sets the axis ticks in the inside of the graph, and on all the edges;
- adds caps to error bars;
- thins the lines of the graph.

**To test:**
Make sure that "build/ExternalData/Testing/Data/UnitTest/" is within the data search directories list, then run 

```
import directtools as dt
from mantid.simpleapi import *

DirectILLCollectData(Run='ILL/IN4/084447', OutputWorkspace='data')
DirectILLReduction(InputWorkspace='data', OutputWorkspace='SofQW', OutputSofThetaEnergyWorkspace='SofTW')

Q = 2.
dQ = 0.2

fig, axes, cuts = dt.plotconstQ('SofQW', Q, dQ)
fig.show()

dosFromStw = ComputeIncoherentDOS('SofTW')
dosFromSqw = ComputeIncoherentDOS('SofQW')
fig, axes = dt.plotDOS([dosFromStw, dosFromSqw], labels=[r'from $S(2\theta, E)$', r'from $S(Q,E)$'])
fig.show()

EMin = -20.
QMax = dt.validQ('SofQW', EMin)[1]
VMax = 0.5 * dt.nanminmax('SofQW')[1]

fig, axes = dt.plotSofQW('SofQW', QMax=QMax, EMin=EMin, VMax=VMax)
fig.show()
```

as a script on Workbench to see plots with these changes.

Fixes #27650 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by Harrietbrown** [comment]
I am having trouble getting the script running, raising this error.

```
Error in execution of algorithm DirectILLCollectData:
The instrument LOQ is not valid for this loader!
  at line 123 in 'C:/Users/wey38795/Documents/mantid_builds/Master/raw/mantid/Framework/PythonInterface/plugins/algorithms\LoadAndMerge.py'
  caused by line 72 in 'C:/Users/wey38795/Documents/mantid_builds/Master/raw/mantid/Framework/PythonInterface/plugins/algorithms\LoadAndMerge.py'
  at line 293 in 'C:/Users/wey38795/Documents/mantid_builds/Master/raw/mantid/Framework/PythonInterface/plugins/algorithms\WorkflowAlgorithms\DirectILLCollectData.py'
  caused by line 737 in 'C:/Users/wey38795/Documents/mantid_builds/Master/raw/mantid/Framework/PythonInterface/plugins/algorithms\WorkflowAlgorithms\DirectILLCollectData.py'
  caused by line 1132 in 'C:\Users\wey38795\Documents\mantid_builds\Master\raw\mantid\Framework\PythonInterface\mantid\simpleapi.py'
RuntimeError: The instrument LOQ is not valid for this loader!
  at line 123 in 'C:/Users/wey38795/Documents/mantid_builds/Master/raw/mantid/Framework/PythonInterface/plugins/algorithms\LoadAndMerge.py'
  caused by line 72 in 'C:/Users/wey38795/Documents/mantid_builds/Master/raw/mantid/Framework/PythonInterface/plugins/algorithms\LoadAndMerge.py'
  at line 293 in 'C:/Users/wey38795/Documents/mantid_builds/Master/raw/mantid/Framework/PythonInterface/plugins/algorithms\WorkflowAlgorithms\DirectILLCollectData.py'
  caused by line 737 in 'C:/Users/wey38795/Documents/mantid_builds/Master/raw/mantid/Framework/PythonInterface/plugins/algorithms\WorkflowAlgorithms\DirectILLCollectData.py'
  caused by line 1132 in 'C:\Users\wey38795\Documents\mantid_builds\Master\raw\mantid\Framework\PythonInterface\mantid\simpleapi.py'
  File "<string>", line 4, in <module>
  File "C:\Users\wey38795\Documents\mantid_builds\Master\raw\mantid\Framework\PythonInterface\mantid\simpleapi.py", line 1132, in algorithm_wrapper
    algm.execute()
```

**Comment by MathieuTillet** [comment]
Well ... the loader seems to mistakenly believe the file comes from the LOQ instrument, and the only way for it to reach this conclusion is to read this in the nexus file itself, under `/instrument/name` ... where obviously it says the file comes from IN4.
So I have to say I have no clue why the script fails this way ...

**Comment by Harrietbrown** [comment]
I found the issue there, I had to change the instrument in my settings. But now I'm getting much more signifficant errors. The first plot works but with a lot of `Warning: converting a masked element to nan.`
but the call of `dosFromStw = ComputeIncoherentDOS('SofTW')` hard crashes mantid.

```
C:\Users\wey38795\Documents\mantid_builds\Master\raw\mantid\external\src\ThirdParty\lib\python2.7\python.exe C:/Users/wey38795/Documents/mantid_builds/Master/build/bin/Debug/workbench-script.pyw
ConfigService-[Information] This is Mantid version 4.2.20200114.915 revision g0764b061dda
ConfigService-[Information] running on NDLT1175 starting 2020-01-14T12:56Z
ConfigService-[Information] Properties file(s) loaded: C:\Users\wey38795\Documents\mantid_builds\Master\build\bin\Debug/Mantid.properties, C:\Users\wey38795\Documents\mantid_builds\Master\build\bin\Debug/Mantid.user.properties
FrameworkManager-[Notice] Welcome to Mantid 4.2.20200114.915
FrameworkManager-[Notice] Please cite: http://dx.doi.org/10.1016/j.nima.2014.07.029 and this release: http://dx.doi.org/10.5286/Software/Mantid
DirectILLCollectData-[Notice] DirectILLCollectData started
DirectILLCollectData-[Notice] Flat background subtraction enabled.
DirectILLCollectData-[Notice] Incident energy calibration enabled.
DirectILLCollectData-[Notice] Applied Ei calibration to '__data_flat_bkg_subtracted_detectors_'.
DirectILLCollectData-[Notice] Original Ei: 9.08935583333 new Ei: 9.13855491878.
DirectILLCollectData-[Notice] Original wavelength: 3.0 new wavelength 2.99191357668.
DirectILLCollectData-[Notice] Applied Ei calibration to '__data_flat_bkg_subtracted_monitors_'.
DirectILLCollectData-[Notice] Original Ei: 9.08935583333 new Ei: 9.13855491878.
DirectILLCollectData-[Notice] Original wavelength: 3.0 new wavelength 2.99191357668.
DirectILLCollectData-[Notice] ElasticChannel set to Fit Elastic Channel by the IPF.
DirectILLCollectData-[Notice] Elastic channel index 174 was used for TOF axis adjustment.
DirectILLCollectData-[Notice] DirectILLCollectData successful, Duration 1.13 seconds
DirectILLReduction-[Notice] DirectILLReduction started
DirectILLReduction-[Notice] Using the default grouping angle step of 0.01 degrees.
DirectILLReduction-[Notice] Binned momentum transfer axis to bin width 0.025 A-1.
DirectILLReduction-[Notice] DirectILLReduction successful, Duration 2.54 seconds
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
Warning: converting a masked element to nan.
ComputeIncoherentDOS-[Notice] ComputeIncoherentDOS started
Rebin2D-[Warning] Fractional area tracking was requested but input workspace does not have calculated bin fractions. Assuming bins are exact (fractions are unity). The results may not be accurate if this workspace was previously rebinned.
Assertion is_set(s) failed on line 207 of file ../../src/tbb/governor.cpp
Detailed description: Attempt to terminate non-local scheduler instance
```

**Comment by Harrietbrown** [comment]
Not the first time I've ran into issues with `ComputeIncoherentDOS`, it does sometihng similar to that in some docs. I'm going to rebuild in release and see what I can get, that worked for building the docs

File: scripts/directtools/__init__.py [hunk 1]
```diff
@@ -234,7 +234,7 @@ def _plotsinglehistogram(workspaces, labels, style, xscale, yscale):
     for ws, label in zip(workspaces, labels):
         if 'm' in style:
             markerStyle, markerIndex = _choosemarker(markers, markerIndex)
-        axes.errorbar(ws, wkspIndex=0, linestyle=lineStyle, marker=markerStyle, label=label, distribution=True)
+        axes.errorbar(ws, wkspIndex=0, linestyle=lineStyle, marker=markerStyle, label=label, distribution=True, capsize=4, linewidth=1)
```
Comment (Harrietbrown) [inline scripts/directtools/__init__.py]: split this over two lines

File: scripts/directtools/__init__.py [hunk 2]
```diff
@@ -626,7 +626,8 @@ def plotcuts(direction, workspaces, cuts, widths, quantity, unit, style='l', kee
                     markerStyle, markerIndex = _choosemarker(markers, markerIndex)
                 realCutCentre, realCutWidth = _cutcentreandwidth(line)
                 label = _label(ws, realCutCentre, realCutWidth, len(workspaces) == 1, len(cuts) == 1, len(widths) == 1, quantity, unit)
```
Comment (Harrietbrown) [inline scripts/directtools/__init__.py]: also while you are in the file this line is too long as well

**Review by Harrietbrown — CHANGES_REQUESTED** [review]
I've gotten the sccript working now in release. That was an issue with the ComputeInchoherantDOS unrelated to this PR.

The plots however look as you describe them. Just keep lines under 120 characters and It s all good.

File: scripts/directtools/__init__.py (hunk 2 above)
Comment (MathieuTillet) [reply scripts/directtools/__init__.py]: Got it ! It should be all good now

**Review by Harrietbrown — APPROVED** [review]
Thats good work.

**Pull request merged**